ci-operator/step-registry/openstack/provision/bastionproxy/openstack-provision-bastionproxy-commands.sh

```diff
@@ -119,33 +136,57 @@ fi
 echo "Deploying squid on $bastion_fip"
 >&2 cat << EOF > $WORK_DIR/deploy_squid.sh
 sudo dnf install -y squid
+
 sudo bash -c "cat << EOF > /etc/squid/squid.conf
 acl localnet src 0.0.0.0/0
 acl SSL_ports port 443
+acl SSL_ports port 53
 acl SSL_ports port 1025-65535
 acl Safe_ports port 80
+acl Safe_ports port 53
 acl Safe_ports port 443
 acl Safe_ports port 1025-65535
 acl CONNECT method CONNECT
 http_access deny !Safe_ports
 http_access deny CONNECT !SSL_ports
 http_access allow localnet
 http_access deny all
-http_port 8213
+http_port 3128
+https_port 3130 cert=/etc/squid/certs/domain.crt key=/etc/squid/certs/domain.key cafile=/etc/squid/certs/domain.crt
+
+# Leave coredumps in the first cache dir
+coredump_dir /var/spool/squid
+
+auth_param basic program /usr/lib64/squid/basic_ncsa_auth /etc/squid/passwd
+auth_param basic children 5
+auth_param basic realm Squid Basic Authentication
+auth_param basic credentialsttl 2 hours
+acl auth_users proxy_auth REQUIRED
+http_access allow auth_users
 EOF"
+
 sudo systemctl start squid
+sudo mkdir -p /etc/squid/certs
+cd /etc/squid/certs
+sudo openssl req -newkey rsa:4096 -nodes -sha256 -keyout domain.key -x509 -days 365 \
+ -addext "subjectAltName = IP:$SUBNET_ID" -subj "/C=US/ST=RH/L=RH/O=RH/CN=shifstack.com" -out domain.crt
+sudo cp /etc/squid/certs/domain.crt /etc/pki/ca-trust/source/anchors/domain.crt
+sudo update-ca-trust
+sudo yum install -y httpd-tools
+htpasswd -bBc /etc/squid/htpasswd dummy dummy
```

Comment:
Idea for later: use secrets to put real credentials here, so malicious users won't be able to use the ephemeral proxy (that as a reminder is open on the Internet :D)